packages/web/src/components/common/my-position-card/MyPositionCard.tsx

```diff
@@ -48,54 +47,54 @@ const MyPositionCard: React.FC<MyPositionCardProps> = ({
-            balanceUSD: existReward.balanceUSD + rewardInfo.balanceUSD,
-            claimableAmount: existReward.claimableAmount + rewardInfo.claimableAmount,
-            claimableUSD: existReward.claimableUSD + rewardInfo.claimableUSD,
-            accumulatedRewardOf1d: existReward.accumulatedRewardOf1d + rewardInfo.accumulatedRewardOf1d,
-            claimableUsdValue: existReward.claimableUsdValue + rewardInfo.claimableUsdValue,
-            aprOf7d: existReward.aprOf7d + rewardInfo.aprOf7d,
-          };
-        } else {
-          infoMap[rewardInfo.rewardType][rewardInfo.token.priceId] = rewardInfo;
-        }
-      });
-    return {
-      SWAP_FEE: Object.values(infoMap["SWAP_FEE"]),
-      STAKING: Object.values(infoMap["STAKING"]),
-      EXTERNAL: Object.values(infoMap["EXTERNAL"]),
-    };
-  }, [rewards, tokenPrices]);
+  //   const infoMap: { [key in RewardType]: { [key in string]: PositionClaimInfo } } = {
+  //     "SWAP_FEE": {},
+  //     "STAKING": {},
+  //     "EXTERNAL": {},
+  //   };
+  //   rewards
+  //     .map(reward => ({
+  //       token: reward.token,
+  //       rewardType: reward.rewardType,
+  //       balance: makeDisplayTokenAmount(reward.token, reward.totalAmount) || 0,
+  //       balanceUSD: Number(reward.totalAmount) * Number(tokenPrices[reward.token.priceId]?.usd || 0),
+  //       claimableAmount: makeDisplayTokenAmount(reward.token, reward.claimableAmount) || 0,
+  //       claimableUSD: Number(reward.claimableUsdValue),
+  //       accumulatedRewardOf1d: makeDisplayTokenAmount(reward.token, reward.accumulatedRewardOf1d || 0) || 0,
+  //       claimableUsdValue: Number(reward.claimableUsdValue),
+  //       aprOf7d: Number(reward.aprOf7d),
+  //     }))
+  //     .forEach((rewardInfo) => {
+  //       const existReward = infoMap[rewardInfo.rewardType][rewardInfo.token.priceId];
+  //       if (existReward) {
+  //         infoMap[rewardInfo.rewardType][rewardInfo.token.priceId] = {
+  //           ...existReward,
+  //           balance: existReward.balance + rewardInfo.balance,
+  //           balanceUSD: existReward.balanceUSD + rewardInfo.balanceUSD,
+  //           claimableAmount: existReward.claimableAmount + rewardInfo.claimableAmount,
+  //           claimableUSD: existReward.claimableUSD + rewardInfo.claimableUSD,
+  //           accumulatedRewardOf1d: existReward.accumulatedRewardOf1d + rewardInfo.accumulatedRewardOf1d,
+  //           claimableUsdValue: existReward.claimableUsdValue + rewardInfo.claimableUsdValue,
+  //           aprOf7d: existReward.aprOf7d + rewardInfo.aprOf7d,
+  //         };
+  //       } else {
+  //         infoMap[rewardInfo.rewardType][rewardInfo.token.priceId] = rewardInfo;
+  //       }
+  //     });
+  //   return {
+  //     SWAP_FEE: Object.values(infoMap["SWAP_FEE"]),
+  //     STAKING: Object.values(infoMap["STAKING"]),
+  //     EXTERNAL: Object.values(infoMap["EXTERNAL"]),
+  //   };
+  // }, [rewards, tokenPrices]);
```

Comment:
Should this area be removed or is it something that is currently difficult to implement ?